### PR TITLE
Mantine menu

### LINKED ISSUE #32273: Menu component
Docs

- [Product doc](https://www.notion.so/metabase/Design-System-Menu-component-f8cffcd6aa374526bf6158d66b66c4f5?pvs=4)
- [Figma component](https://www.figma.com/file/MZhwfwmOaa8HeCBBUCeq7R/Menu?type=design&node-id=5-70&mode=design&t=4iBnf5NgLe2uH7PQ-11)

```[tasklist]
### Tasks
- [ ] Export the [Mantine](https://mantine.dev/core/menu/) `Menu` component from the `metabase/frontend/src/ui` according to the component’s category.
- [ ] Make sure the component style matches the component in Figma. You can add component specific styling to the `components:` section of `frontend/src/metabase/ui/theme.ts`
- [ ] Create a storybook file for the new component in the same directory. You can use `frontend/src/metabase/ui/components/inputs/Radio/Radio.stories.mdx` as an example.

```

### PR DESCRIPTION
**Pull request by ranquild**
Closes https://github.com/metabase/metabase/issues/32273

Changes:
- Exports `Menu` and adds style overrides
- Exports `Button` for testing, yet to be styled in a separate PR
- Changes `spacing`, `radius`, `shadows` values to what we use in the project, and updates overrides 
- Moves component overrides to the component folders

How to verify:
- Run `yarn storybook`
- Open http://localhost:6006/?path=/story/layovers-menu--with-label-and-divider
- Make sure it works according to the figma file https://www.figma.com/file/MZhwfwmOaa8HeCBBUCeq7R/Menu?type=design&node-id=5-70&mode=design&t=1cSNiSttrRRUaZGV-0

Screenshot 2023-08-11 at 16 09 48

### CODE REVIEW

File: frontend/src/metabase/ui/components/layovers/Menu/Menu.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+  },
+  closeOnEscape: {
+    control: { type: "boolean" },
+  },
+  closeOnItemClick: {
+    control: { type: "boolean" },
+  },
+};
+
+<Meta title="Layovers/Menu" component={Menu} args={args} argTypes={argTypes} />
```
Comment (kdoh) [inline frontend/src/metabase/ui/components/layovers/Menu/Menu.stories.mdx:60]:
```suggestion
<Meta title="Overlays/Menu" component={Menu} args={args} argTypes={argTypes} />
```
Would tweak this so it matches our Figma structure.

File: frontend/src/metabase/ui/components/layovers/Menu/Menu.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+</Canvas>
+
+export const WithRightSection = args => (
+  <Flex justify="center">
+    <Menu {...args}>
+      <Menu.Target>
+        <Button>Toggle menu</Button>
+      </Menu.Target>
+      <Menu.Dropdown>
+        <Menu.Item>Text</Menu.Item>
```
Comment (kdoh) [inline frontend/src/metabase/ui/components/layovers/Menu/Menu.stories.mdx:113]:
```suggestion
        <Menu.Item>Option 1</Menu.Item>
```

File: frontend/src/metabase/ui/components/layovers/Menu/Menu.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        <Menu.Item>Text</Menu.Item>
```
Comment (kdoh) [inline frontend/src/metabase/ui/components/layovers/Menu/Menu.stories.mdx:114]:
```suggestion
        <Menu.Item>Option 2 - Electric Boogaloo</Menu.Item>
```

File: frontend/src/metabase/ui/components/layovers/Menu/Menu.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        <Menu.Item>Text</Menu.Item>
```
Comment (kdoh) [inline frontend/src/metabase/ui/components/layovers/Menu/Menu.stories.mdx:115]:
```suggestion
        <Menu.Item>Option 3</Menu.Item>
```

File: frontend/src/metabase/ui/components/layovers/Menu/Menu.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+        <Menu.Item rightSection={<Text color="inherit">⌘K</Text>}>
+          Text
```
Comment (kdoh) [inline frontend/src/metabase/ui/components/layovers/Menu/Menu.stories.mdx:117]:
```suggestion
          Keyboard option
```

File: frontend/src/metabase/ui/components/layovers/Menu/Menu.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+        </Menu.Item>
+        <Menu.Item>Text</Menu.Item>
```
Comment (kdoh) [inline frontend/src/metabase/ui/components/layovers/Menu/Menu.stories.mdx:119]:
```suggestion
        <Menu.Item>Option 5</Menu.Item>
```

File: frontend/src/metabase/ui/components/layovers/Menu/Menu.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+</Canvas>
+
+export const WithIcon = args => (
+  <Flex justify="center">
+    <Menu {...args}>
+      <Menu.Target>
+        <Button>Toggle menu</Button>
+      </Menu.Target>
+      <Menu.Dropdown>
+        <Menu.Item icon={<Icon name="dashboard" />}>Text</Menu.Item>
```
Comment (kdoh) [inline frontend/src/metabase/ui/components/layovers/Menu/Menu.stories.mdx:136]:
```suggestion
        <Menu.Item icon={<Icon name="question" />}>Question</Menu.Item>
```

File: frontend/src/metabase/ui/components/layovers/Menu/Menu.stories.mdx [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+        <Menu.Item icon={<Icon name="dashboard" />}>Text</Menu.Item>
```
Comment (kdoh) [inline frontend/src/metabase/ui/components/layovers/Menu/Menu.stories.mdx:137]:
```suggestion
        <Menu.Item icon={<Icon name="dashboard" />}>SQL Query</Menu.Item>
```

File: frontend/src/metabase/ui/components/layovers/Menu/Menu.stories.mdx [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+        <Menu.Item icon={<Icon name="dashboard" />}>Text</Menu.Item>
```
Comment (kdoh) [inline frontend/src/metabase/ui/components/layovers/Menu/Menu.stories.mdx:138]:
```suggestion
        <Menu.Item icon={<Icon name="dashboard" />}>Dashboard</Menu.Item>
```

File: frontend/src/metabase/ui/components/layovers/Menu/Menu.stories.mdx [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+        <Menu.Item icon={<Icon name="dashboard" />}>Text</Menu.Item>
```
Comment (kdoh) [inline frontend/src/metabase/ui/components/layovers/Menu/Menu.stories.mdx:139]:
```suggestion
        <Menu.Item icon={<Icon name="dashboard" />}>Collection</Menu.Item>
```

File: frontend/src/metabase/ui/components/layovers/Menu/Menu.stories.mdx [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+        <Menu.Item icon={<Icon name="dashboard" />}>Text</Menu.Item>
```
Comment (kdoh) [inline frontend/src/metabase/ui/components/layovers/Menu/Menu.stories.mdx:140]:
```suggestion
        <Menu.Item icon={<Icon name="dashboard" />}>Action</Menu.Item>
```

**Review by kdoh — CHANGES_REQUESTED** [review]
Looking good in general. I do think it would be worthwhile to add more "realistic" data in the previews to better sense how things are working. I provided some examples as suggestions.

I'd like Dawei to take a look as well and confirm that he's happy with the spacing and sizing since he was in charge of the spec for this one. One thing I noticed was that the hover color felt a little bit too dark compared to what's in product in spots like the `New` menu. I know it matches "what's in Figma" but curious if we have any different feelings seeing it in its native habitat.

**Comment by dhuniverse** [comment]
Looking good in general. A couple of small udpates to consider. 

1.  On the docs tab / General section, the [Toggle menu] was inside a bounding div. Clicking on it won't be able to show the entire menu list. I would suggest showing the menu popup totally via moving the [Toggle menu] out of the bounding area, or remove the bounding constraints.
2. On the docs tab / General section, the sub-sections are identical. Suggestion adding a sub-header to differentiate the sub-sections.
3. For the top sections on the docs tab, I have written a draft content for each. Please see those in [Figma](https://www.figma.com/file/MZhwfwmOaa8HeCBBUCeq7R/Menu?type=design&node-id=442%3A676&mode=design&t=SggFFFXFkpkiTfBA-1).

**Review by kdoh — COMMENTED** [review]
Happy to move forward once Dawei's happy with things.

**Comment by ranquild** [comment]
@dhuniverse I've updated the docs, all the mentioned issues should be fixed. Please take a look

Screenshot 2023-08-15 at 12 33 28

**Review by oisincoveney — COMMENTED** [review]
Looks really good from my side, love that you broke out the styles as well! I'm happy if the others are happy

**Review by kdoh — APPROVED** [review]
Nice work. I'm happy. Let's ship it. 🥇  (Also very into the extracted overrides).

**Pull request merged**